### DISCUSSION TITLE
[sign-client v2] Question about disconnecting and throwing an error

### DISCUSSION THREAD

**Discussion by juliancwirko** [WalletConnect/walletconnect-monorepo#1810] [Developer Support]
Hello, could someone please explain why an error is thrown when disconnecting? I mean, `No matching key. session or pairing topic doesn't exist: {topic}`. I would like to understand the purpose of throwing that error on each disconnection. Is it because there is no other way to signal it differently using `ping`? (I mean that this was the manual disconnection for a purpose, and there is no need to throw an error). Thanks.

I saw that it is also tested here: https://github.com/WalletConnect/walletconnect-monorepo/blob/v2.0/packages/sign-client/test/sdk/client.spec.ts#L71.

**Comment by daniellnog** [comment]
@juliancwirko did you figure out the reason for this issue? 

I'm having a similar issue. Since I started my development integration with v2.0, I've created lot of sessions. Then, In my app, I'm trying to delete all of them, but, the most recents I can disconnect, but, old sessions throws me the same error

**Reply by juliancwirko** [reply]
Unfortunately, nothing new on my side. I am curious to know whether it is expected behavior or not.

**Reply by daniellnog** [reply]
> Unfortunately, nothing new on my side. I am curious to know whether it is expected behavior or not.

:( Sad! Will try to understand it...let me know on this post in case you have found a way to solve this.

**Reply by daniellnog** [reply]
@juliancwirko I solve my problem. Since I had create many sessions, the wallet connect library uses local storage to store some info, so, sometimes you are connected but the current session looks like not the “current”, and to solve that, I just did a clean up on all local storages related to wc@ and then, the error is gone! Let me know if I can help you

**Reply by deantek** [reply]
@daniellnog I do the same, but it seems to me that this is not the best solution

**Comment by finessevanes** [comment]
Hi @juliancwirko, reviewing old discussions here. Is this issue persisting? If so, please try the latest version. Can we close this discussion if it's resolved?

**Reply by juliancwirko** [reply]
Hi, I still see it in v2.6.2. Unfortunately, I can't test the newest version for now.

**Reply by finessevanes** [reply]
Thanks for letting me know. Checking in with our team to see if this has been resolved in the newest version.

**Reply by finessevanes** [reply]
Moving this into an [issue](https://github.com/WalletConnect/walletconnect-monorepo/issues/2529) and closing this discussion. Further comments will be made on the issue itself.
